File: ePrijevozSarajevo/ePrijevozSarajevo.Services/Database/Vehicle.cs
namespace ePrijevozSarajevo.Services.Database
{
    public class Vehicle
    {
        public int VehicleId { get; set; }
        public int Number { get; set; }
        public string RegistrationNumber { get; set; } = null!;
        public int BuildYear { get; set; }
        
        public Manufacturer? Manufacturer { get; set; }
        public int ManufacturerId { get; set; }
        public Type? Type { get; set; }
        public int TypeId { get; set; }

       // public virtual ICollection<VehicleManufacturer> VehicleManufacturers { get; set; } = new List<VehicleManufacturer>();
        //public virtual ICollection<VehicleType> VehicleTypes { get; set; } = new List<VehicleType>();
    }
}

File: ePrijevozSarajevo/UI/eprijevoz_desktop/lib/models/vehicle.dart
import 'package:json_annotation/json_annotation.dart';
part 'vehicle.g.dart';

@JsonSerializable()
class Vehicle {
  int? vehicleId;
  int? number;
  String? registrationNumber;
  int? buildYear;
  int? manufacturerId;
  int? typeId;

  Vehicle({this.vehicleId, this.registrationNumber, this.buildYear});

  factory Vehicle.fromJson(Map<String, dynamic> json) =>
      _$VehicleFromJson(json);

  Map<String, dynamic> toJson() => _$VehicleToJson(this);
}
